src/grpcbox_stream.erl

```diff
@@ -209,6 +217,10 @@ on_receive_data(Bin, State=#state{request_encoding=Encoding,
     catch
         throw:{grpc_error, {Status, Message}} ->
             end_stream(Status, Message, State);
+        throw:{grpc_extended_error, #{status := Status, message := Message} = ErrorData} ->
+            Trailers = maps:get(trailers, ErrorData, #{}),
+            State2 = update_trailers(maps:to_list(Trailers), State),
+            end_stream(Status, Message, State2);
```

Comment:
Make into a separate function to be used in `handle_info('EXIT'...` as where the `maps:get` had the map and key transposed.